toolchain/sem_ir/inst.h

```diff
@@ -305,11 +305,15 @@ inline auto operator<<(llvm::raw_ostream& out, TypedInst inst)
 // Associates a LocId and Inst in order to provide type-checking that the
 // TypedNodeId corresponds to the InstT.
 struct LocIdAndInst {
+  // Used when there is no associated location. Note, we should generally do our
+  // best to associate a location for diagnostics.
```

Comment:
```suggestion
  // Constructs a LocIdAndInst with no associated location. Note, we should generally do our
  // best to associate a location for diagnostics.
```

I find function comments much clearer when they focus primarily on the function's contract, i.e. what it does or what it returns, rather than how it's used.